lib/spack/spack/repo.py

```diff
@@ -684,8 +684,15 @@ def from_descriptors(
     @staticmethod
     def from_config(config: spack.config.Configuration) -> "RepoPath":
         """Create a RepoPath from a configuration object."""
+
+        def replace_vars_in_package_attrs(attrs):
+            for a in ("git", "url"):
+                if a in attrs:
+                    attrs[a] = spack.util.path.substitute_path_variables(attrs[a])
+            return attrs
```

Comment:
I don't love the selective list here. I also think that if we are going to have a selective list then I think it should be an error to try and expand something not in the list.

It should also be added to the documentation so users know what the rules are.

Reply:
We generally select which vars are path-type vars to apply `spack.util.path.substitute_path_variables`, we don't really have an automatic component to that.

I'm not sure why we couldn't just apply it to all values read from config -- there's nowhere that you need to use those variables without expansion that I can think of

Reply:
I like that idea. We have so many one off's where the configs break due to expansion. I think I average 1 per year for a config missing expansions.

Reply:
@psakievich @becker33 Implemented the substitution for the entire config and worked my way through the unit test failures. There were a few gotchas. Not sure if there is anything more I can do for the import check.

Reply:
Thanks @rbberger. Not sure how I missed this. I'll review tomorrow